src/BaGet.Core/Services/PackageIndexingService.cs

```diff
@@ -32,7 +32,12 @@ public class PackageIndexingService : IPackageIndexingService
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
         }
 
-        public async Task<PackageIndexingResult> IndexAsync(Stream packageStream, CancellationToken cancellationToken)
+        public async Task<PackageIndexingResult> IndexAsync(
+            Stream packageStream,
+            bool storage = true,
```

Comment:
Please remove the booleans you added. `IndexAsync` should be idempotent. If you call `IndexAsync` for a package you've already indexed, absolutely nothing should happen: it should no-op without throwing any exceptions. For example, `FileStorageService` allows you to upload the same package to storage twice (see [this code](https://github.com/loic-sharma/BaGet/blob/master/src/BaGet.Core/Services/FileStorageService.cs#L76)). This means that you should be able to blindly call `IndexAsync` on any package you find in the drop folder. Some notes:

1. Currently, idempotency is broken on AWS S3 and GCP. We should fix that...
2. Calling `IndexAsync` twice on the same package should "reindex" the package and reupload it to storage and reindex it in search. I wouldn't worry about this now as this should be out of scope for this pull request